src/models/whatsnewmodel.cpp

```diff
@@ -0,0 +1,148 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "whatsnewmodel.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "settingsholder.h"
+#include "featurelist.h"
+#include "feature.h"
+
+#include <QList>
+
+namespace {
+Logger logger(LOG_MODEL, "WhatsNewModel");
+}  // namespace
+
+WhatsNewModel::WhatsNewModel() { MVPN_COUNT_CTOR(WhatsNewModel); }
+
+WhatsNewModel::~WhatsNewModel() { MVPN_COUNT_DTOR(WhatsNewModel); }
+
+QList<Feature*> m_featurelist;
+bool m_initialized = false;
+
+void maybeInitialize() {
+  if (m_initialized) {
+    return;
+  }
+
+  m_initialized = true;
+
+  QList<Feature*> allFeatures = Feature::getAll();
+  QList<Feature*> newFeatures;
+
+  for (int i = 0; i < allFeatures.count(); ++i) {
+    bool shouldBeInWhatsNew = allFeatures[i]->isNew() &&
+                              allFeatures[i]->isMajor() &&
+                              allFeatures[i]->isSupported();
+
+    if (shouldBeInWhatsNew) {
+      newFeatures.append(allFeatures[i]);
+    }
+  }
+
+  m_featurelist = newFeatures;
+}
+
+int WhatsNewModel::featureCount() const {
```

Comment:
Remove `const` otherwise you are not allowed changing the object members. The object must be in readonly mode for const methods. So, remove this const and all should work.